lib/controller.py

```diff
@@ -521,3 +523,97 @@ def do_dun(self, line):
     def do_undun(self, line):
         results = self.cluster.infoUndun(self.mods['line'], nodes=self.nodes)
         self.view.dun(results, self.cluster, **self.mods)
+
+@CommandHelp('"collectinfo" is used to collect system stats on the local node.')
+class CollectinfoController(CommandController):
+    def collect_local_file(self,src,dest_dir):
+        shutil.copy2(src, dest_dir)
+        
+    def write_log(self,collectedinfo,src_file='/var/log/aerospike/*log'):
+        aslogdir = '/tmp/as_log_' + str(time.time())
+        aslogfile = aslogdir + '/ascollectinfo.log'
+        os.mkdir(aslogdir)
+        f = open(str(aslogfile), 'w')
+        f.write(str(collectedinfo))
+        f.close()
+        self.collect_local_file(src_file,aslogdir)
+        util.shell_command(["tar -czvf " + aslogdir + ".tgz " + aslogdir])
+        sys.stderr.write("\x1b[2J\x1b[H")
+        print "\n\n\nFiles in " + aslogdir + " and " + aslogdir + ".tgz saved. Please send tgz archive to support@aerospike.com"
+        print "END OF ASCOLLECTINFO"
+
+    def do_collectinfo(self, line):
+        capture_stdout = util.capture_stdout
+        collect_output = ''
```

Comment:
yes, this was more erring on side of precaution, just in there is a failure later on. fixed it now.